horace_core/utilities/compute_bin_data.m

```diff
@@ -0,0 +1,91 @@
+function [mean_signal, mean_variance,signal_msd] = compute_bin_data( ...
+    npix,signal,variance,no_normalization)
+% Compute the mean signal and variance given the number of contributing
+% pixels for each bin
+% Returns empty arrays if obj contains no pixels.
+%
+%   >> [mean_signal, mean_variance,std_deviations] = compute_bin_data(npix,signal,variance)
+%   >> [mean_signal, mean_variance,std_deviations] = compute_bin_data(npix,signal,variance,no_normalization)
+%
+% Input
+% -----
+% npix      The number of contributing pixels to each bin of the plot axes
+% signal    The signal array, to calculate binning for
+%
+% Optional:
+% variance  The variance array to calculate binning
+%
+% no_normalization -- if present and true, do not normalize signal and
+%            variance by npix to calculate proper averages. Return sums
+%            only
+%
+% Output
+% ------
+% mean_signal     The average signal for each bin.
+%                 size(mean_signal) = size(npix)
+% Optional:
+%
+% mean_variance   The average variance for each bin.
+%                 size(mean_variance) = size(npix)
+%
+% signal_msd      If mean_variance is calculated from signal,
+%                 signal_std contans the mean square deviation of each
+%                 element of signal arry in a bin from its mean value in
+%                 this bin. If variance is present, it returned in signal_std
+%
+%
+if nargin<3
+    variance = [];
+end
+if nargin<4
+    normalize  = true;
+else
+    normalize  = ~no_normalization;
+end
+if nargout > 1
+    calc_variance= true;
+else
+    calc_variance= false;
+end
+if nargout > 2
+    calc_signal_msd = true;
+else
+    calc_signal_msd = false;
+end
```

Comment:
Can just use `calc_signal_msd = nargout > 2`

Reply:
will add this to the next ticket